### ISSUE TITLE
Numpy 1.14 depreciation warnings

### ISSUE THREAD

**Issue by thomasaarholt** [hyperspy/hyperspy#1817]
With the latest numpy (1.14), there are a few depreciation warnings being raised:
hyperspy/io.py is raising a warning about line 341:
```python
hyperspy\hyperspy\io.py:341: FutureWarning: 
Conversion of the second argument of issubdtype from `complex` 
to `np.complexfloating` is deprecated. In future, it will be treated as 
`np.complex128 == np.dtype(complex).type`.
  if np.issubdtype(dtype, complex):
```
I suspect we want to replace `complex` with `np.complexfloating`, if I understand the warning correctly.
Variations of `np.issubdtype(dtype, complex)` (and float) are also present in `complex_signal.py`, `eds.py` and `signal.py`.

h5py is also raising a warning:
```python
FutureWarning: Conversion of the second argument of issubdtype from 
`float` to `np.floating` is deprecated. 
In future, it will be treated as `np.float64 == np.dtype(float).type`.
```
It is being [fixed](https://github.com/h5py/h5py/pull/926) ( not yet merged)

**Comment by ericpre** [comment]
This is already fixed by #1799 and #1806. Since the other warnings are not related to HyperSpy, I think we can close this?

**Comment by thomasaarholt** [comment]
Great to hear! I hadn't realised.
